File: src/main/java/com/epicsquid/whoosh/gui/ListBoxPointElement.java
package com.epicsquid.whoosh.gui;

import cofh.core.client.gui.element.listbox.ListBoxElementText;
import com.epicsquid.whoosh.utils.TransporterPoint;

public class ListBoxPointElement extends ListBoxElementText {

	private final TransporterPoint point;

	public ListBoxPointElement(TransporterPoint point) {
		super(point.dim());
		this.point = point;
	}


	@Override
	public TransporterPoint getValue() {
		return point;
	}
}

File: src/main/java/com/epicsquid/whoosh/gui/ListBoxPointElement.kt
package com.epicsquid.whoosh.gui

import cofh.core.client.gui.element.listbox.ListBoxElementText
import com.epicsquid.whoosh.utils.TransporterPoint

class ListBoxPointElement(private val point: TransporterPoint) : ListBoxElementText(point.dim()) {
    override fun getValue(): TransporterPoint {
        return point
    }
}
